### PR TITLE
expand: fix minor diagnostics bug

### PR DESCRIPTION
**Pull request by Xiretza**
The error mentions `///`, when it's actually `//!`:

```
error[E0658]: attributes on expressions are experimental
 --> test.rs:4:9
  |
4 |         //! wah
  |         ^^^^^^^
  |
  = note: see issue https://github.com/rust-lang/rust/issues/15701 <https://github.com/rust-lang/rust/issues/15701> for more information
  = help: add `#![feature(stmt_expr_attributes)]` to the crate attributes to enable
  = help: `///` is for documentation comments. For a plain comment, use `//`.
```

### CODE REVIEW

File: compiler/rustc_expand/messages.ftl [hunk 1]
```diff
@@ -10,6 +10,11 @@ expand_attribute_meta_item =
 expand_attribute_single_word =
     attribute must only be a single word
 
+expand_attributes_on_expressions_experimental =
+    attributes on expressions are experimental
+    .help_outer_doc = `///` is for documentation comments. For a plain comment, use `//`.
+    .help_inner_doc = `//!` is for documentation comments. For a plain comment, insert a space: `// !`
```
Comment (compiler-errors) [inline compiler/rustc_expand/messages.ftl:16]: can you make this say "use // or insert a space"? i think it's likely that the ! could be erroneous too

**Review by compiler-errors — APPROVED** [review]

File: compiler/rustc_expand/messages.ftl (hunk 1 above)
Comment (estebank) [inline compiler/rustc_expand/messages.ftl:16]:
```suggestion
    .help_outer_doc = `///` is used for outer documentation comments; for a plain comment, use `//`.
    .help_inner_doc = `//!` is used for inner documentation comments; for a plain comment, `//` by removing the `!` or inserting a space in between them `// !`
```

**Review by estebank — APPROVED** [review]
Could you add a test triggering these two cases?

File: compiler/rustc_expand/messages.ftl [hunk 2]
```diff
@@ -10,6 +10,11 @@ expand_attribute_meta_item =
 expand_attribute_single_word =
     attribute must only be a single word
 
+expand_attributes_on_expressions_experimental =
+    attributes on expressions are experimental
+    .help_outer_doc = `///` is used for outer documentation comments; for a plain comment, use `//`.
```
Comment (compiler-errors) [inline compiler/rustc_expand/messages.ftl:15]:
```suggestion
    .help_outer_doc = `///` is used for outer documentation comments; for a plain comment, use `//`
```

nit

Comment (Xiretza) [reply compiler/rustc_expand/messages.ftl]: good spot, thanks

**Comment by Xiretza** [comment]
ping

**Comment by Xiretza** [comment]
I don't *think* that was my fault? Not entirely sure though, maybe the relevant logs are just somewhere else.

**Pull request merged**